routes/auth.js

```diff
@@ -0,0 +1,111 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+
+const router = express.Router();
+const User = require('../models/User');
+const passport = require('passport');
+
+//Documentation for Swagger https://github.com/fliptoo/swagger-express 
+
+/**
+ * @swagger
+ * /:
+ *  get:
+ *    description: use to request authentication page
+ *    responses:
+ *       '200': 
+ *       description: Successfully   
+ *       
+ */
+
+
+router.get('/auth/signup', (req, res) => {
+  res.render('auth/signup');
+});
+
+/**
+ * @swagger
+ * /:
+ *  post:
+ *    description: use to post authentication page
+ *    responses:
+ *       '200': 
+ *       description: Successfully   
+ *       
+ */
+
+
+router.post(
+  '/auth/login',
+  passport.authenticate('local', {
+    successRedirect: '/',
+    failureRedirect: '/auth/login',
+    failureFlash: true,
+    passReqToCallback: true
+  })
+)
+
+// github authentication routing
+router.get('/auth/github', passport.authenticate('github'));
+
+router.get(
+  '/auth/github/callback',
+  passport.authenticate('github', {
+    successRedirect: '/',
+    failureRedirect: '/auth/login',
+  })
+);
+
+
```

Comment:
Add annotations  from Swagger for Documentation Routes

Reply:
Done